### ISSUE TITLE
Ubuntu 18 installation problems

### ISSUE THREAD

**Issue by nateGeorge** [freqtrade/freqtrade#2107]
## Step 1: Have you search for this issue before posting it?

If you have discovered a bug in the bot, please [search our issue tracker](https://github.com/freqtrade/freqtrade/issues?q=is%3Aissue). 
If it hasn't been reported, please create a new issue.

## Step 2: Describe your environment

  * Operating system: Ubuntu 18.04.2 LTS
  * Python Version: Python 3.6.9 :: Anaconda custom (64-bit) (`python -V`)
  * CCXT version: _____ (`pip freeze | grep ccxt`)
  * Branch: Master | Develop
  * Last Commit ID: _____ (`git log --format="%H" -n 1`)
  
## Step 3: Describe the problem:

First, when building there seems to be a missing package.  Once this is done:

`sudo apt install libc6-dev-i386`

it is able to install.  Found from here: https://github.com/xoreaxeaxeax/rosenbridge/issues/5

Part of the error was `bits/libc-header-start.h: No such file or directory`

Then, once that was fixed, `lzma` could not be found.  It is looking within the system python3.7 libraries for some reason.  It seems that python3.7 has this lzma issue.  Running

`python3.7`
`import lzma`

Seems to have the same error of `No module named _lzma.`.

With python3.6 it seems to work.  I ended up deleting the check for python3.7 in the setup.sh file and it worked.  However, it seems I had some weird installation of python3.7 in my system, and it didn't have lzma installed with it.  So I ended up deleting the python executable at /usr/local/bin/python3.7, and reinstalling python3.7 with `sudo apt install python3.7`, then install pip with these steps: https://stackoverflow.com/a/56140616/4549682

Lastly, I was using postgresql, and it needed the `psycopg2` package installed with pip before that would work.

### Steps to reproduce:

  1. Fresh Ubuntu 18.04 install
  2. Try to install freqtrade
  
### Observed Results:

  * What happened?
  * What did you expect to happen?

### Relevant code exceptions or logs:

  ```
bits/libc-header-start.h: No such file or directory
  ```

**Comment by nateGeorge** [comment]
This is just documentation for these problems in case anyone else encounters them, so closing the issue.

**Comment by xmatthias** [comment]
I'll reopen this so we don't forget to add this to the documentation.

**Comment by hroff-1902** [comment]
> I'll reopen this so we don't forget to add this to the documentation.

How can this be documented? It looks like yet another shamanistic ritual...

**Comment by xmatthias** [comment]
well recreate this environment , test installation ... also the postgres stuff should be documented (we know since > 1 year that that will work, but the docs don't contain the word postgres ...)

**Comment by nateGeorge** [comment]
Agree, the psycopg2 could be installed in the virtualenv by default (it doesn't take that much space) and the `sudo apt install libc6-dev-i386` seems to be necessary for Ubuntu 18 for the TAlib build to work.  Maybe could be added to the setup.sh file or whereever dependencies are installed.

**Comment by xmatthias** [comment]
I don't understand the dependency you're missing (`libc6-dev-i386`) - it isn't required on my (brandnew vagrant/virtualbox) setup.

using a brand-new vagrant box ( Ubuntu 18.04.3 LTS - 64bit), running the setup.sh script (after installing python3-pip and python3-venv) does run through, installing ta-lib correctly.

the only libc6 i have installed are `libc6:amd64` and `libc6-dev:amd64`.

**Comment by nateGeorge** [comment]
Ah, perhaps it's the two libc6 packages you mentioned that are an issue.  Possibly those were uninstalled or never installed on mine.  When I do `apt list --installed | grep libc6` I get:

```bash
libc6/bionic,now 2.27-3ubuntu1 amd64 [installed]
libc6-dbg/bionic,now 2.27-3ubuntu1 amd64 [installed]
libc6-dev/bionic,now 2.27-3ubuntu1 amd64 [installed]
libc6-dev-i386/bionic,now 2.27-3ubuntu1 amd64 [installed]
libc6-dev-x32/bionic,now 2.27-3ubuntu1 amd64 [installed,automatic]
libc6-i386/bionic,now 2.27-3ubuntu1 amd64 [installed,automatic]
libc6-x32/bionic,now 2.27-3ubuntu1 amd64 [installed,automatic]
```

Not sure why the x32 is installed (assuming that's 32-bit); I'm running a 64-bit version.

**Comment by xmatthias** [comment]
`libc6-i386` is the 32bit version .. x32 seems to be there for cross-compiling stuff (found [here](https://packages.debian.org/sid/libc6-x32-cross).

I'm not sure on fixing this as part of setup.sh - since it seems very specific to your compiler setup ...

**Comment by nateGeorge** [comment]
Agreed, probably just leave it in this issue in case it's needed for anyone else.

**Comment by hroff-1902** [comment]
Closing it, feel free to reopen if still have concern.
